armi/reactor/converters/uniformMesh.py

```diff
@@ -1100,13 +1101,45 @@ def _mapStateFromReactorToOther(
                     aDest,
                     self.paramMapper,
                     mapNumberDensities,
-                    calcReactionRates=self.calcReactionRates,
+                    calcReactionRates=False,
+                )
+
+            # If requested, the reaction rates will be calculated based on the
+            # mapped neutron flux and the XS library.
+            if self.calcReactionRates:
+                self._calculateReactionRatesEfficient(
+                    destReactor.core, sourceReactor.core.p.keff
                 )
 
         # Clear the cached data after it has been mapped to prevent issues with
         # holding on to block data long-term.
         self._cachedReactorCoreParamData = {}
 
+    @staticmethod
+    def _calculateReactionRatesEfficient(core, keff):
+        """
+        First, sort blocks into groups by XS type. Then, we just need to grab micros for each XS type once.
+
+        Iterate over list of blocks with the given XS type; calculate reaction rates for these blocks
+        """
+        from armi.physics.neutronics.globalFlux.globalFluxInterface import (
+            calcReactionRatesBlockList,
+        )
```

Comment:
My major question here is if it is worthwhile to have the `calcReactionRatesBlockList()` method as a part of the API. It feels quite specific to me, and I'm not sure if it is worth being public.

If we instead acknowledged the limited use cases of `calcReactionRatesBlockList()` by maybe making it private, we could clean things up in a few ways:
1. You can remove this import within a method
2. You could remove a number of `if` statements that are there to catch edge cases, which would speed things up even more
3. You could stop passing around so many arguments in function calls

My feeling is that it could just be a lot cleaner overall, and that adding `calcReactionRatesBlockList()` to the API won't benefit hardly anybody.

Reply:
I can move `calcReactionRatesBlockList` to uniformMesh.py (the only place it is used) and then make it private, but I don't immediately see how that's going to help with (2) and (3). I'll push up what I have and maybe you can point out the opportunities for simplification.

Reply:
I made some suggestions in comments. I might be off base, but hopefully they make sense.

I also wonder if the abstraction from blocks to `objList` is worth anything. I guess maybe, I don't know. Not a big deal.